Source/SwiftLintBuiltInRules/Rules/Lint/UnusedDeclarationRule.swift

```diff
@@ -27,6 +27,7 @@ struct UnusedDeclarationRule: ConfigurationProviderRule, AnalyzerRule, Collectin
         name: "Unused Declaration",
         description: "Declarations should be referenced at least once within all files linted",
         kind: .lint,
+        minSwiftVersion: .fiveDotEight,
```

Comment:
I don't think this needs to be a breaking change, since the rule works fine on older Swift versions if you're not using shorthand optional bindings.